spec-tests/src/test/java/org/candlepin/spec/pools/PoolSpecTest.java

```diff
@@ -0,0 +1,625 @@
+            .owner(Owners.toNested(owner))
+            .type(ctype);
+
+        if (facts != null) {
+            // Copy the map into a new hashmap to ensure we can still modify it later if we so choose
+            consumer.facts(new HashMap<>(facts));
+        }
+
+        consumer.putFactsItem("system.certificate_version", "3.3");
+
+        return client.consumers()
+            .register(consumer);
+    }
+
+    private ProductDTO createProduct(ApiClient client, OwnerDTO owner) throws ApiException {
+        ProductDTO product = Products.randomSKU();
+
+        return client.ownerProducts()
+            .createProductByOwner(owner.getKey(), product);
+    }
+
+    // ugh...
+    private AttributeDTO buildAttribute(String key, String value) {
+        return new AttributeDTO()
+            .name(key)
+            .value(value);
+    }
+
+    private PoolDTO createPool(ApiClient client, OwnerDTO owner) throws ApiException {
+        ProductDTO product = this.createProduct(client, owner);
+        return this.createPool(client, owner, product);
+    }
+
+    private PoolDTO createPool(ApiClient client, OwnerDTO owner, ProductDTO product) throws ApiException {
+        PoolDTO pool = Pools.randomUpstream(product);
+
+        return client.owners()
+            .createPool(owner.getKey(), pool);
+    }
+
+
+    private ApiClient createUserClient(ApiClient client, OwnerDTO owner) throws ApiException {
+        UserDTO user = UserUtil.createUser(client, owner);
+        return ApiClients.trustedUser(user.getUsername(), true);
+    }
+
+    private ApiClient createConsumerClient(ApiClient client, OwnerDTO owner) throws ApiException {
+        String consumerName = StringUtil.randomSuffix("test_consumer");
+        ConsumerDTO consumer = this.createConsumer(client, owner, consumerName, null);
+
+        return this.createConsumerClient(client, consumer);
+    }
+
+    private ApiClient createConsumerClient(ApiClient client, ConsumerDTO consumer) throws ApiException {
+        return ApiClients.ssl(consumer.getIdCert());
+    }
+
+
+    @Test
+    @DisplayName("should let consumers view their own pools")
+    public void shouldLetConsumersViewTheirOwnPools() throws ApiException {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = this.createOwner(adminClient);
+        PoolDTO pool = this.createPool(adminClient, owner);
+
+        ApiClient userClient = this.createUserClient(adminClient, owner);
+
+        List<PoolDTO> pools = userClient.pools()
+            .listPools(owner.getId(), null, null, true, null);
+
+        assertNotNull(pools);
+        assertEquals(1, pools.size());
+
+        ApiClient consumerClient = this.createConsumerClient(userClient, owner);
+
+        PoolDTO fetched = consumerClient.pools()
+            .getPool(pool.getId(), null, null);
+
+        assertNotNull(fetched);
+        assertEquals(pool.getId(), fetched.getId());
+    }
+
+  // it 'includes null values in json' do
```

Comment:
same here: can we remove the ruby test comment since it's still in the .rb file?

Reply:
done